pkg/graveler/graveler.go

```diff
@@ -371,27 +371,30 @@ type CommitParams struct {
 type KeyValueStore interface {
 	// Get returns value from repository / reference by key, nil value is a valid value for tombstone
 	// returns error if value does not exist
-	Get(ctx context.Context, repositoryID RepositoryID, ref Ref, key Key) (*Value, error)
+	Get(ctx context.Context, repository *RepositoryRecord, ref Ref, key Key) (*Value, error)
```

Comment:
I think we should aim at replacing the `RepositoryID` with `*RepositoryRecord` just for the operations that are part of the `ListCommits` hot path. The Graveler API will become less coherent for the meantime, but it will allow us to quick fix this without performing a massive change to this big API.

Reply:
Would keep these changes - It will improve all calls to Graveler and it is more than one call that will require to have a dual interface/implementation or anomaly that accepts repository information.